### PR TITLE
introduce a cmdctrl  k command called report bug

### LINKED ISSUE #49585: Introduce a Cmd/Ctrl + K command called "File a bug"
People should also be able to report a bug via the Cmd/Ctrl + K command.
Flow:
1. Cmd or Ctrl + K
2. Type "bu" or "report" (or whatever else matches "Report a bug")
3. Select "Report a bug"
4. Pop the diagnostics dialog

### PR DESCRIPTION
**Pull request by filiphric**
Closes https://github.com/metabase/metabase/issues/49585 and https://github.com/metabase/metabase/issues/49584

### Description

Add Diagnostic data window to command palette and user settings menu

### How to verify

Describe the steps to verify that the changes are working as expected.

Search functionality
1. cmd + k
2. search for "report a bug".  also can by found by searching  words like: bug, report, issue, problem, error, diagnostic

User menu
1. open user menu
2. click on "report a bug" item

### Demo

Screenshot 2024-11-13 at 13 19 04
Screenshot 2024-11-13 at 13 18 52

### CODE REVIEW

**Review by mazameli — CHANGES_REQUESTED** [review]
I asked @romnitt just now in Figma if we could use a better icon for the report-a-bug action rather than the `info` one.

(Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)

**Review by rafpaf — CHANGES_REQUESTED** [review]
The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?

**Comment by filiphric** [comment]
> The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?

wouldn’t unit test make more sense here?

**Comment by filiphric** [comment]
I updated the icon and added a test to cover the scenario. I opted in for the e2e test, since I could not figure out an effective way of how to write a unit test 😅

**Comment by rafpaf** [comment]
> > The interface and code look spiffing! Could you please add e2e tests for finding "Report a bug" in both places?
> 
> wouldn’t unit test make more sense here?

Yes, good call - unit tests would be fine and arguably even better.

**Comment by thebiglabasky** [comment]
> (Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)

Right I hadn't thought of that. I wonder if we could trick that by using "File a bug" instead 😅?

**Comment by filiphric** [comment]
> > (Separately, more of a comment or small worry than a change request: I wonder if @thebiglabasky [what a good GH handle] is concerned about people seeing this action too often when they're just searching for a dashboard or saved question that has the name `Report` in it. Will that give people too much of an impression that Metabase is so buggy that that action needs to pop up frequently?)
> 
> Right I hadn't thought of that. I wonder if we could trick that by using "File a bug" instead 😅?

I changed the text from "report a bug" to "file a bug" and removed "report" as a keyword so that it would not appear in the search too often. should I change this in the profile link menu as well?

**Comment by thebiglabasky** [comment]
> I changed the text from "report a bug" to "file a bug" and removed "report" as a keyword so that it would not appear in the search too often. should I change this in the profile link menu as well?

Nice, I'd say so indeed to stay consistent

**Comment by filiphric** [comment]
I updated this. it’s ready to merge once I get an approval

**Comment by thebiglabasky** [comment]
There are quite some people here in the review list, maybe it's easier if we had one specific person? Also noting that this is Thanksgiving so unlikely for north american colleagues to jump in until Monday. Would anyone elsewhere be available?

**Review by uladzimirdev — APPROVED** [review]
Code itself LGTM, nice work.

As per product/design perspective, I'd double check if we need to show actions before the search results. I doubt users are looking for a "report bug" action often.

UPD: I see @mazameli had the same concern

**Comment by filiphric** [comment]
@rafpaf @mazameli I implemented the changes can you re-review please?

**Review by rafpaf — APPROVED** [review]

**Review by npfitz — CHANGES_REQUESTED** [review]
I believe you can already open that modal through the command palette via "Open error diagnostic modal".

Can we just change the action name?

**Comment by filiphric** [comment]
> I believe you can already open that modal through the command palette via "Open error diagnostic modal".
> 
> Can we just change the action name?

you’re right, I did not know about this. I pushed changes to fix this.

File: e2e/test/scenarios/onboarding/command-palette.cy.spec.js
Comment (npfitz) [inline e2e/test/scenarios/onboarding/command-palette.cy.spec.js:251]: The functionality here would already be covered in https://github.com/metabase/metabase/blob/49585-introduce-a-cmdctrl--k-command-called-report-bug/e2e/test/scenarios/admin-2/error-reporting.cy.spec.ts#L48. Can we update this test to reflect the new name?

**Review by npfitz — CHANGES_REQUESTED** [review]
As mentioned, I think there is an existing e2e test that can be adjusted for the Command Palette action. Can we also get some coverage on the new item in the drop down menu here?
image

File: frontend/src/metabase/nav/components/ProfileLink/ProfileLink.jsx
Comment (npfitz) [inline frontend/src/metabase/nav/components/ProfileLink/ProfileLink.jsx:169]: Something else is that this modal already exists in the React Component tree in `App.tsx` as `<KeyboardTriggeredErrorModal />`. Rather than have it here twice, can the "Filke a bug" action just dispatch `openDiagnostics()` as well? Rather than have the modal in the component tree twice?

**Review by mazameli — COMMENTED** [review]
Changes seem good to me — the only weirdness is that you select an action that says "File a bug," and then you see a modal that says "download diagnostic info." I know that the plan is that we're going to update that modal so that it'll make sense with this new action — has the modal already been updated in `master` but just not in this branch?

image

**Comment by filiphric** [comment]
> Changes seem good to me — the only weirdness is that you select an action that says "File a bug," and then you see a modal that says "download diagnostic info." I know that the plan is that we're going to update that modal so that it'll make sense with this new action — has the modal already been updated in `master` but just not in this branch?
> 
> image

yes there’s a slight discrepancy in the wording. the good news is that we are slowly moving away from simply downloading information towards actually reporting a bug. this is the PR that is coming right after this one: https://github.com/metabase/metabase/pull/50277 and there’s a [whole project](https://www.notion.so/metabase/Quality-initiatives-update-14d69354c90180648b0fce2c0268e856?pvs=4) that will improve this flow. the "File a bug" will make much more sense in upcoming weeks.

**Comment by mazameli** [comment]
@filiphric Not to be pedantic, but then shouldn't https://github.com/metabase/metabase/pull/50277 target _this_ branch, and then once it's integrated, we merge the whole together into `master`? Typically when something is merged to `master` it means we think it's complete. cc @thebiglabasky

**Comment by filiphric** [comment]
> @filiphric Not to be pedantic, but then shouldn't #50277 target _this_ branch, and then once it's integrated, we merge the whole together into `master`? Typically when something is merged to `master` it means we think it's complete. cc @thebiglabasky

It’s okay to be pedantic, I’m still trying to grasp how things work around here. my approach was to work granularly and finish small tasks to push the big thing out in small chunks. I’m realizing this was probably not the best approach, since it takes a little while to push things into production. I’ll definitely change my style for the future, because right now I created a train of PRs piggybacking on top of each other and the whole experience was a pain. my PRs are stuck in review for 4 weeks, I’m pinging devs for reviews and switching contexts all day. I have 5 PRs now, all seem to be green so hopefully I can push them out of the way.

**Comment by filiphric** [comment]
closed in favor of https://github.com/metabase/metabase/pull/50856

**Pull request closed without merging**
